sql: fill in functionality to CREATE DATABASE with REGIONS

otan opened the pull request:
Release note (sql change): `CREATE DATABASE ... WITH REGIONS ...`
now allows the creation of database with region metadata.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55764)
<!-- Reviewable:end -->

otan committed:
sql: fill in functionality to CREATE DATABASE with REGIONS

Release note (sql change): `CREATE DATABASE ... WITH REGIONS ...`
now allows the creation of database with region metadata.

otan commented:
superseded by #56131
